ppr-ui/src/composables/mhrRegistration/useHomeOwners.ts

```diff
@@ -438,13 +433,55 @@ export function useHomeOwners (isMhrTransfer: boolean = false) {
 
       Object.assign(groupToUpdate, {
         ...fractionalData,
-        ...(hasChanges && { action: ActionTypes.CHANGED })
+        ...(hasChanges && { action: isMhrTransfer ? ActionTypes.CHANGED : ActionTypes.CORRECTED })
       })
 
       const updatedOwnerGroups = [...homeOwnerGroups]
       setTransferOrRegistrationHomeOwnerGroups(updatedOwnerGroups)
     }
 
+  /**
+   * Return the baseline owners snapshot by id.
+   * @param ownerId The owner identifier
+   */
+  const getMhrBaselineOwnerById = (ownerId: number): MhrRegistrationHomeOwnerIF => {
+    let currentOwner: MhrRegistrationHomeOwnerIF
+    getMhrBaseline.value?.ownerGroups.forEach(group => group.owners.find(owner => {
+      if (owner.ownerId === ownerId) currentOwner = owner
```

Comment:
Short circuit with a `return` statement?

Reply:
Sorry, i don't follow. Could you elaborate what you mean?

Reply:
Sure

Reply:
Refactored for maximum speed, hopefully improves the speed at which the owner is returned.